### ISSUE TITLE
Doesn't train with shuffled labels

### ISSUE THREAD

**Issue by wuyifat** [SeanNaren/deepspeech.pytorch#354]
The model trained fine as long as I kept the underscore at the first place in the label, but WER stayed at 98 when I also shuffled the underscore.

I did remember to pass in the blank_index
```python
labels = ''.join(random.sample(labels, len(labels)))
decoder = GreedyDecoder(labels, blank_index=labels.index('_'))
```

Anyone knows where the bug could be?

**Comment by ryanleary** [comment]
The CTC loss function assumes the blank character is first.
